### DISCUSSION TITLE
Rebase and Merge

### DISCUSSION THREAD

**Discussion by jamesaorson** [panix-os-dev-team/Panix#169] [General]
Why are we using merge commits instead of rebase and merge? I have found that rebase and merge makes a much cleaner commit history and makes merge conflicts much easier to fix.

**Comment by micahswitzer** [comment]
The general strategy I was working off of was: once your PR is approved, you rebase on the target branch to ensure you have incorporated the most recent changes and resolved any conflicts that might exist. You then would be able to cleanly merge the changes in.

I'm not sure if this is what you were suggesting or not, but I like to keep the single merge to show where changes were brought in to the main branch.

**Reply by Kfeavel** [reply]
I agree with this. I like the distinction that an actual merge provides, but all branches should be rebased on top of `dev` or whatever branch is appropriate before it is merged in.

**Reply by jamesaorson** [reply]
Gotcha. What I do to maintain distinction is making every commit be prefixed with the issue number, like so `123: Some changes`, so you can grep for all commits related to an issue.

**Reply by Kfeavel** [reply]
That's not a bad idea, but I already struggle with making my messages fit within the 50 character limit, lol.

**Reply by micahswitzer** [reply]
That's why commits can span multiple lines. The issue number could really appear anywhere in the message.

**Reply by Kfeavel** [reply]
Fair enough -- I have no problem with doing that from here on out. I do think rebasing and then merging is still the way to go though.
